src/simplewallet/simplewallet.cpp

```diff
@@ -873,65 +873,86 @@ bool simple_wallet::set_log(const std::vector<std::string> &args)
 //----------------------------------------------------------------------------------------------------
 bool simple_wallet::ask_wallet_create_if_needed()
 {
+  LOG_PRINT_L3("simple_wallet::ask_wallet_create_if_needed() started");
   std::string wallet_path;
-
-  bool valid_path = false;
-  do {
-    wallet_path = command_line::input_line(
-        tr("Specify wallet file name (e.g., MyWallet). If the wallet doesn't exist, it will be created.\n"
-        "Wallet file name: ")
-    );
-    if (std::cin.eof())
-    {
-      return false;
-    }
-    valid_path = tools::wallet2::wallet_valid_path_format(wallet_path);
-    if (!valid_path)
-    {
-      fail_msg_writer() << tr("wallet file path not valid: ") << wallet_path;
-    }
-  }
-  while (!valid_path);
-
+  std::string confirm_creation;
+  bool wallet_name_valid = false;
   bool keys_file_exists;
   bool wallet_file_exists;
-  tools::wallet2::wallet_exists(wallet_path, keys_file_exists, wallet_file_exists);
-  LOG_PRINT_L3("wallet_path: " << wallet_path << "");
-  LOG_PRINT_L3("keys_file_exists: " << std::boolalpha << keys_file_exists << std::noboolalpha
-    << "  wallet_file_exists: " << std::boolalpha << wallet_file_exists << std::noboolalpha);
-
-  LOG_PRINT_L1("Loading wallet...");
 
-  // add logic to error out if new wallet requested but named wallet file exists
-  if (keys_file_exists || wallet_file_exists)
-  {
-    if (!m_generate_new.empty() || m_restoring)
+  do{
+      LOG_PRINT_L3("Request user input");
+      wallet_path = command_line::input_line(
+      tr("Specify wallet file name (e.g., MyWallet). If the wallet doesn't exist, it will be created.\n"
+         "Wallet file name (or Ctrl-C to quit): ")
+       );
+    if(std::cin.eof())
     {
-      fail_msg_writer() << tr("attempting to generate or restore wallet, but specified file(s) exist.  Exiting to not risk overwriting.");
+      LOG_ERROR("Unexpected std::cin.eof() - Exited simple_wallet::ask_wallet_create_if_needed()");
       return false;
     }
-  }
-
-  bool r;
-  if(keys_file_exists)
-  {
-    m_wallet_file=wallet_path;
-    r = true;
-  }else
-  {
-    if(!wallet_file_exists)
+    if(wallet_path.empty()) //Check for empty name
     {
-      std::cout << tr("The wallet doesn't exist, generating new one") << std::endl;
-      m_generate_new = wallet_path;
-      r = true;
-    }else
+      fail_msg_writer() << tr("Empty wallet names are invalid. Please try again or use Ctrl-C to quit.");
```

Comment:
Because the entire function reads as follows:

```
bool wallet2::wallet_valid_path_format(const std::string& file_path)
{
  return !file_path.empty();
}
```
So I now just directly check for `if(wallet_path.empty()) //Check for empty name` on line `894`